### PR TITLE
complete tests for three subcomponent. 

### PR DESCRIPTION
**Pull request by quanjingchen**
1. make axios call and render the component within beforeEach
2. use async and await for async events like axios call, button click etc...
3. use jest.fn() to mock function

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ import TrackerContext from '../utilities/TrackerContext.js';
 
 function App() {
   const trackClick = useContext(TrackerContext);
-  const [productID, setProductID] = useState(40352);
+  const [productID, setProductID] = useState(40349);
```
Comment (tjspitz) [inline client/src/components/App.jsx:18]: - possibly trivial comment, but, we should have team chat about which ID we want to default to for presentation & deployment
- (nothing to change right now)

File: client/src/components/App.jsx [hunk 2]
```diff
@@ -42,6 +42,9 @@ function App() {
 
   return (
     <div id="global">
+      <div id="Head" onClick={trackClick}>
```
Comment (tjspitz) [inline client/src/components/App.jsx:45]: - also not an "action now" but need to have a consensus on whether to use ids for each major component, or to always use classNames
- originally it was to use ids, but, I think right now there's a mix

File: client/src/components/Sub_Head/styles/head.css [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  background: transparent;
+  border: none;
+  outline: none;
+  margin-left: -33px;
+  /* margin-top: 1rem; */
+}
+
+.icon-magnifying-glass {
+  font-size: 1.2rem;
+}
```
Comment (tjspitz) [inline client/src/components/Sub_Head/styles/head.css:54]: Power hour w/ css, nice!

**Review by tjspitz — APPROVED** [review]
Overall comments/notes:

- did not go line by line through the `coverage` files, hope that's ok
- consider slightly more descriptive img alt text? but not a big blocker for me
  * e.g. `main product ${product.name}` instead of just `"main product"`
- as always, fantastic work!

**Comment by quanjingchen** [comment]
i like this idea! "e.g. main product ${product.name} instead of just "main product" will do it.

**Pull request merged**